### PR TITLE
Updates to schedule repository

### PR DESCRIPTION
**Pull request by dephillipsmichael**
The schedule repository now can update the schedule from a task result and task run UUID.  It also now supports failed updates to bridge being retried at a later time until they are successful. 

@liujoshua is working on integrating the ScheduleRespoitory.shared.updateSchedule(taskResult, taskRunUuid) call into the BoundServiceTaskResultManager

### CODE REVIEW

File: sageresearch-app-sdk/src/main/java/org/sagebionetworks/research/sageresearch/viewmodel/ScheduleRepository.kt [hunk 1]
```diff
@@ -176,20 +226,206 @@ open class ScheduleRepository(context: Context) {
...
+                }, {
+                    scheduleOnMain(it, onError)
+                }))
+    }
+
+    /**
+     * Helper function to schedule calls on the main thread
+     */
+    @VisibleForTesting
+    protected open fun scheduleOnMain(
```
Comment (dephillipsmichael) [inline sageresearch-app-sdk/src/main/java/org/sagebionetworks/research/sageresearch/viewmodel/ScheduleRepository.kt:328]: Is there a better way to do this?

File: sageresearch-app-sdk/src/main/java/org/sagebionetworks/research/sageresearch/viewmodel/ScheduleRepository.kt [hunk 2]
```diff
@@ -176,20 +226,206 @@ open class ScheduleRepository(context: Context) {
...
+        scheduleTaskRunUuidMap[uuid] = schedule.guid
+        return uuid
+    }
+
+    /**
+     * @param schedule that we are going to update on bridge
+     *                 schedule can be null if the user does a task without internet
+     * @param taskResult the result of the user doing the schedule task
+     */
+    fun updateSchedule(taskResult: TaskResult, taskRunUuid: UUID) {
```
Comment (dephillipsmichael) [inline sageresearch-app-sdk/src/main/java/org/sagebionetworks/research/sageresearch/viewmodel/ScheduleRepository.kt:262]: @liujoshua you would call this function after the TaskResult is completed

Comment (liujoshua) [reply sageresearch-app-sdk/src/main/java/org/sagebionetworks/research/sageresearch/viewmodel/ScheduleRepository.kt:262]: Cool, do you want me to do this integration?

File: sageresearch-app-sdk/src/main/java/org/sagebionetworks/research/sageresearch/viewmodel/ScheduleRepository.kt (hunk 1 above)
Comment (liujoshua) [reply sageresearch-app-sdk/src/main/java/org/sagebionetworks/research/sageresearch/viewmodel/ScheduleRepository.kt:328]: Is this what you're thinking of?

```observable
.operatorX // called on IO thread, even though subscribeOn comes afterwards in call chain
.subscribeOn(Schedulers.io()) // subscribe thread actually can only be set once (and where you call it doesn't change the behavior)
.operatorY // called on IO thread
.observeOn(AndroidSchedulers.mainThread()) //switches subsequent operators to main thread
.operatorZ // called on main thread

**Comment by dephillipsmichael** [comment]
I'm getting a bunch of crazy git errors that I can't figure out since you merged in develop.  I'm going to close and re-open a new PR

File: sageresearch-app-sdk/src/main/java/org/sagebionetworks/research/sageresearch/viewmodel/ScheduleRepository.kt (hunk 1 above)
Comment (dephillipsmichael) [reply sageresearch-app-sdk/src/main/java/org/sagebionetworks/research/sageresearch/viewmodel/ScheduleRepository.kt:328]: Hmmm it's not letting me call observeOn after a subscribeOn

Comment (liujoshua) [reply sageresearch-app-sdk/src/main/java/org/sagebionetworks/research/sageresearch/viewmodel/ScheduleRepository.kt:328]: checked out this branch (on your updated PR) taking a look

**Pull request closed without merging**